ext/datasource/etcdv3/etcdv3.go

```diff
@@ -0,0 +1,150 @@
+package etcdv3
+
+import (
+	"context"
+	"fmt"
+	"github.com/alibaba/sentinel-golang/ext/datasource"
+	"github.com/alibaba/sentinel-golang/logging"
+	"github.com/alibaba/sentinel-golang/util"
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/mvcc/mvccpb"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	"time"
+)
+
+var logger = logging.GetDefaultLogger()
+
+type Etcdv3DataSource struct {
+	datasource.Base
+	propertyKey         string
+	lastUpdatedRevision int64
+	client              *clientv3.Client
+	// cancel is the func, call cancel will stop watching on the propertyKey
+	cancel context.CancelFunc
+	// closed indicate whether continuing to watch on the propertyKey
+	closed util.AtomicBool
+}
+
+// NewDatasource new a Etcdv3DataSource instance.
+// client is the etcdv3 client, it must be useful and should be release by User.
+func NewDatasource(client *clientv3.Client, key string, handlers ...datasource.PropertyHandler) (*Etcdv3DataSource, error) {
+	if client == nil {
+		return nil, errors.New("The etcdv3 client is nil.")
+	}
+	ds := &Etcdv3DataSource{
+		client:      client,
+		propertyKey: key,
+	}
+	for _, h := range handlers {
+		ds.AddPropertyHandler(h)
+	}
+	return ds, nil
+
+}
+
+func (s *Etcdv3DataSource) ReadSource() ([]byte, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	resp, err := s.client.Get(ctx, s.propertyKey)
+	if err != nil {
+		return nil, err
+	}
+	if resp.Count == 0 {
+		return nil, errors.Errorf("The key[%s] is not existed in etcd.", s.propertyKey)
+	}
+	s.lastUpdatedRevision = resp.Header.GetRevision()
+	return resp.Kvs[0].Value, nil
+}
+
+func (s *Etcdv3DataSource) Initialize() error {
+	err := s.doReadAndUpdate()
+	if err != nil {
+		return err
```

Comment:
Here we might need to properly handle the error (e.g. parse error) in case of getting in the way of initialization.